src/main/java/org/jenkinsci/plugins/p4/trigger/P4Trigger.java

```diff
@@ -110,15 +109,15 @@ public File getLogFile() {
 	}
 
 	private boolean matchServer(Job<?, ?> job, String port) {
-		if (job instanceof AbstractProject) {
-			AbstractProject<?, ?> project = (AbstractProject<?, ?>) job;
-			SCM scm = project.getScm();
-
-			if (scm instanceof PerforceScm) {
-				PerforceScm p4scm = (PerforceScm) scm;
+		//Get all the trigger for this Job
+		SCMTriggerItem item = SCMTriggerItem.SCMTriggerItems.asSCMTriggerItem(job);
+		//As soon as we find a match, return
+		for (SCM scmTrigger : item.getSCMs()) {
+			if (scmTrigger instanceof PerforceScm) {
+				PerforceScm p4scm = (PerforceScm) scmTrigger;
 				String id = p4scm.getCredential();
 				P4BaseCredentials credential = ConnectionHelper.findCredential(id);
-				if (port.equals(credential.getP4port())) {
+				if (credential.getP4port().equals(port)) {
```

Comment:
Unclear how this swap is related to the topic of the PR, but FindBugs considers it unsafe.